pkg/daemon/update.go

```diff
@@ -2083,13 +2113,28 @@ func (dn *Daemon) experimentalUpdateLayeredConfig() error {
 			return err
 		}
 
-		_, err = client.RebaseLayered(desiredImage)
+		_, err = client.RebaseLayered(desiredImage, !ctrlcommon.InSlice(postConfigChangeActionReboot, actions))
 		if err != nil {
 			return err
 		}
 
 		//defer os.Unlink("/run/ostree/auth.json")
 
+		// Ideally we would want to update kernelArguments only via MachineConfigs.
+		// We are keeping this to maintain compatibility and OKD requirement.
+		if err := UpdateTuningArgs(KernelTuningFile, CmdLineFile); err != nil {
+			return err
+		}
+
+		if err := dn.performPostConfigChangeAction(actions, desiredImage); err != nil {
+			return err
+		}
+
+		if dn.nodeWriter != nil {
```

Comment:
Do we need to check if this is nil? Looks like we do it in `setWorking` but not `updateConfigAndState`. I think we also might need to add some more of the code from `updateConfigAndState`